anndata/_core/merge.py

```diff
@@ -141,7 +141,13 @@ def equal_dask_array(a, b) -> bool:
 
 @equal.register(np.ndarray)
 def equal_array(a, b) -> bool:
-    return equal(pd.DataFrame(a), pd.DataFrame(asarray(b)))
+    # Reshaping allows us to compare inputs with >2 dimensions
+    # We cast to pandas since it will still work with non-numeric types
+    b = asarray(b)
+    if a.shape != b.shape:
+        return False
+
+    return equal(pd.DataFrame(a.reshape(-1)), pd.DataFrame(b.reshape(-1)))
```

Comment:
What does `_.reshape(-1)` mean? The numpy docs say

> One shape dimension can be -1. In this case, the value is inferred from the length of the array and remaining dimensions.

So what you do basically creates 1D `pd.DataFrames`? Shouldn’t we make and compare `pd.Series` then?

Reply:
That would make sense, but in practice the Series constructor acts differently and throws errors for some of the non-numeric dtypes.